lib/Mojolicious/Plugin/OpenAPI.pm

```diff
@@ -308,34 +319,30 @@ sub _register_plugins {
 }
 
 sub _render {
-  my ($renderer, $c, $output, $options) = @_;
-
-  # fallback to default renderer
-  unless (exists $c->stash->{openapi}) {
-    my $renderer = $c->app->renderer;
-    my $handler  = $renderer->handlers->{$renderer->default_handler};
-    $c->app->log->debug(
-      "Using default_handler to render data since 'openapi' was not found in stash. Set 'handler' in stash to avoid this message."
-    );
-    local $options->{handler} = $renderer->default_handler;
-    return $renderer->$handler($c, $output, $options);
-  }
+  my ($renderer, $c, $output, $args) = @_;
+  return unless exists $c->stash->{openapi};
+  return unless my $self = _self($c);
 
-  my $self = _self($c) or return;
-  my $status = $c->stash('status') || 200;
-  my $res = $c->stash('openapi');
+  my $res     = $c->stash('openapi');
+  my $status  = $args->{status} ||= ($c->stash('status') || 200);
+  my $op_spec = $c->openapi->spec || {responses => {$status => {schema => $self->_default_schema}}};
+  my @errors;
 
+  delete $args->{encoding};
   $c->stash->{format} ||= 'json';
-  delete $options->{encoding};
 
-  if (my @errors = $self->validator->validate_response($c, $c->openapi->spec, $status, $res)) {
-    $self->_log($c, '>>>', \@errors);
-    $c->stash(status => 500);
-    $$output = $self->_renderer->($c, {errors => \@errors, status => 500});
+  if ($op_spec->{responses}{$status} or $op_spec->{responses}{default}) {
+    @errors = $self->validator->validate_response($c, $op_spec, $status, $res);
+    $args->{status} = 500 if @errors;
   }
   else {
-    $$output = $self->_renderer->($c, $res);
+    $args->{status} = 501;
```

Comment:
Changed from 500 to 501 if the response code is not defined in the spec, which will hopefully make the situation in #80 less confusing.